integration-tests/src/test/java/oracle/kubernetes/operator/ItJrfModelInImage.java

```diff
@@ -0,0 +1,243 @@
+    assertDoesNotThrow(() -> DbUtils.deleteRcuPod(getResultDir()),
+        "Failed: drop RCU schema");
+    assertDoesNotThrow(() -> DbUtils.deleteDbPod(getResultDir()),
+        "Failed: stop DB");
+  }
+
+  /**
+   * Releases k8s cluster lease, archives result, pv directories.
+   */
+  @AfterAll
+  public static void staticUnPrepare() {
+    assertDoesNotThrow(() -> tearDown(new Object() {
+        }.getClass().getEnclosingClass().getSimpleName(), namespaceList.toString()),
+         "tearDown failed");
+  }
+
+  /**
+   * Create and deploy a JRF domain using model in image. Save and restore walletFileSecret. After shutting down 
+   * the domain, enable walletFileSecret in the domain yaml file. Restart the domain with the modified domain yaml 
+   * file that reuses the same RCU schema. Verify in the restarted domain the required pods, services are created 
+   * and the servers are ready.
+   */
+  @Test
+  public void testReuseRCU2Deployments() {
+    Assumptions.assumeTrue(QUICKTEST);
+    String testMethodName = new Object() {
+    }.getClass().getEnclosingMethod().getName();
+    assertDoesNotThrow(() -> logTestBegin(testMethodName));
+    
+    boolean testCompletedSuccessfully = false;
+    try {
+      Map<String, Object> domainMap =
+          createModelInImageMap(getNewSuffixCount(), testClassName);
+      domainMap.put("namespace", domainNS);
+      domainMap.put("wdtModelFile", "./model.jrf.yaml");
+      domainMap.put("wdtModelPropertiesFile", "./model.properties");
+      domainMap.put("domainHomeImageBase", BaseTest.getfmwImageName() + ":" + BaseTest.getfmwImageTag());
+      domainMap.put("rcuSchemaPrefix", rcuSchemaPrefix);
+      LoggerHelper.getLocal().log(Level.INFO, "DEBUG " + testClassName + "domain: dbUrl: " + dbUrl);
+      domainMap.put("rcuDatabaseURL", dbUrl);
+      domainMap.put("wdtDomainType", "JRF");
+      domainMap.put("introspectorJobActiveDeadlineSeconds", "300");
+      
+      String domainUid = (String)domainMap.get("domainUID");
+      String namespace = (String)domainMap.get("namespace");
+      
+      //create rcuAccess secret and walletPassword secret
+      Secret rcuAccess = assertDoesNotThrow(() -> new RcuSecret(namespace, domainUid + "-rcu-access", 
+          rcuSchemaPrefix, rcuSchemaPass, dbUrl));
+      LoggerHelper.getLocal().log(Level.INFO, "RCU access secret is createdfor {0}", domainUid);
+      
+      Secret walletPass = assertDoesNotThrow(() -> new WalletPasswordSecret(namespace, 
+          domainUid + "-opss-wallet-password-secret", walletPassword));
+      LoggerHelper.getLocal().log(Level.INFO, "Wallet password secret is created for {0}", domainUid);
+      
+      jrfdomain = assertDoesNotThrow(() -> new JrfDomain(domainMap), 
+          "Failed: JRF domain creation");
+      LoggerHelper.getLocal().log(Level.INFO, "JRF domain is created for {0}", testClassName);
+      assertDoesNotThrow(() -> jrfdomain.verifyDomainCreated(40), 
```

Comment:
magic number

Reply:
/**
   * Verifies the required pods are created, services are created and the servers are ready.
   *
   * @param maxIterations max iteration count
   * @throws RuntimeException if pods/services of the domain are not created or WLS is not running
   */
  public void verifyDomainCreated(int maxIterations) throws Exception 

So here 40 is max iteration count. Right now from the tests results it seems to be good but I might need to turn it later after more external Jenkins runs. We have another overloaded method 

public void verifyDomainCreated() throws Exception {
    verifyDomainCreated(BaseTest.getMaxIterationsPod());
 }
40 is the return value from BaseTest.getMaxIterationsPod().

I agree that 40 is a magic number :)
In fact I have JIRA https://jira.oraclecorp.com/jira/browse/OWLS-81246 : Modify the MaxIteration/Loop Count for the tests to reasonable values